synapse/handlers/message.py

```diff
@@ -978,6 +982,44 @@ async def handle_new_client_event(
             await self.store.remove_push_actions_from_staging(event.event_id)
             raise
 
+    async def _cache_joined_hosts_for_event(self, event: EventBase):
+        """Precalculate the joined hosts at the event, when using Redis, so that
+        external federation senders don't have to recalculate it themselves.
+        """
```

Comment:
```suggestion
    async def _cache_joined_hosts_for_event(self, event: EventBase) -> None:
        """Precalculate the joined hosts at the event, when using Redis, so that
        external federation senders don't have to recalculate it themselves.
        """
```

I find this docstring a bit confusing too, but couldn't figure out a good way to reword it.

Also should this be public (no `_` prefix) since it is called from another handler?

Reply:
Yes it should be public